FXThemes/src/main/java/com/pixelduke/window/ThemeWindowManagerFactory.java

```diff
@@ -4,25 +4,46 @@
 import javafx.stage.Window;
 
 public abstract class ThemeWindowManagerFactory {
+    
     public static ThemeWindowManager create() {
         String osName = System.getProperty("os.name");
-        // Check which operating system we're running and return appropriate ThemeWindowManager
-        if (Platform.isWindows()) {
-            if (osName.equals("Windows 10")) {
-                return new Win10ThemeWindowManager();
-            } else {
-                return new Win11ThemeWindowManager();
-            }
-        } else if (Platform.isMac()) {
-            return new MacThemeWindowManager();
-        } else if (Platform.isLinux()) {
-            return new LinuxThemeWindowManager();
-        } else {
-            return new ThemeWindowManager(){
-                public void setDarkModeForWindowFrame(Window window, boolean darkMode){
-                    System.out.println("Warning: Unsupported Window Operating System");
+        int majorVersion = getMajorVersion();
+        
+        // Check which operating system and version we're running and return appropriate ThemeWindowManager
+        switch(Platform.getOSType()){
+            case Platform.WINDOWS:
+                switch(majorVersion){
+                    case 10:
+                        return new Win10ThemeWindowManager();
+                    case 11:
+                        return new Win11ThemeWindowManager();
+                    default:
+                        break;
                 }
-            };
+                break;
+            case Platform.MAC:
+                return new MacThemeWindowManager();
+            case Platform.LINUX:
+                return new LinuxThemeWindowManager();
+            default:
+                break;
+        }
+        
+        return new ThemeWindowManager(){
+            public void setDarkModeForWindowFrame(Window window, boolean darkMode){
+                System.out.println("Warning: Unsupported Window Operating System");
+            }
+        };
+    }
+
+    private static int getMajorVersion(){
+        int version = -1;
+        
+        try{
+            version = Integer.parseInt(System.getProperty("os.version").split("\\.")[0]);
+        }catch (Exception e){
+            e.printStackTrace();
```

Comment:
I think it would be better to also print some more information here. Something like "we couldn't parse the version number of the OS..."?

Reply:
I could do that if desired. I just added it for kicks and error handling tbh. It should NEVER fail, but never say never right? I'll throw in a little `System.err` print out rq.

Reply:
Changes pushed!